core/src/main/scala/com.snowplowanalytics.elasticsearch/loader/NsqSourceExecutor.scala

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2014-2016 Snowplow Analytics Ltd.
+ * All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0,
+ * and you may not use this file except in compliance with the Apache
+ * License Version 2.0.
+ * You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the Apache License Version 2.0 is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied.
+ *
+ * See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+
+package com.snowplowanalytics.elasticsearch.loader
+
+//NSQ
+import com.github.brainlag.nsq.NSQConsumer
+import com.github.brainlag.nsq.lookup.DefaultNSQLookup
+import com.github.brainlag.nsq.NSQMessage
+import com.github.brainlag.nsq.callbacks.NSQMessageCallback
+
+//Java
+import java.nio.charset.StandardCharsets.UTF_8
+
+// Scalaz
+import scalaz._
+import Scalaz._
+
+// This project
+import sinks._
+import clients._
+import StreamType._
+
+/**
+ *
+ * @param streamType the type of stream, good/bad
+ * @param documentIndex the elasticsearch index name
+ * @param documentType the elasticsearch index type
+ * @param nsqConfig the NSQ configuration
+ * @param goodSink the configured GoodSink
+ * @param badSink the configured BadSink
+ * @param elasticsearchSender function for sending to elasticsearch
+ */
+class NsqSourceExecutor(
+  streamType: StreamType,
+  documentIndex: String,
+  documentType: String,
+  nsqConfig: ElasticSinkNSQConfig,
+  goodSink: Option[ISink],
+  badSink: ISink,
+  elasticsearchSender: ElasticsearchSender
+) extends Runnable {
+      // Use same transformer with the stdin
+      // because records are coming in same format basically
+      val transformer = streamType match {
+        case StreamType.Good => new SnowplowElasticsearchTransformer(documentIndex, documentType)
+        case StreamType.Bad => new BadEventTransformer(documentIndex, documentType)
+      }
+
+      val topicName = streamType match {
+          case StreamType.Good => nsqConfig.nsqGoodSourceTopicName
+          case StreamType.Bad => nsqConfig.nsqBadTopicName
+      }
+
+      def run = {
```

Comment:
please reformat the signature to

```scala
override def run(): Unit
```

parens + Unit return type generally means side-effects so it's always good to flag up